### PR TITLE
Fixes Bounding/Clicking Bugs

### PR DESCRIPTION
**Pull request by clusterfack**
This fixes a number of bugs with the clicking system.

- Wearable animated components were fixed so that they properly report their AABB 
- Wearable animated components draw their AABB in the AABB debug
- Wearable animated components were fixed so that they properly use their nonwornsprite or wornsprite for checkclicking
- Fixes the clicking system only choosing the player character
- Fixes the eraser only working within a 1.5 tile range

### CODE REVIEW

File: SS14.Client/GameObjects/ClientEntityManager.cs [hunk 1]
```diff
@@ -53,6 +53,26 @@ public IEnumerable<IEntity> GetEntitiesIntersecting(Box2 position)
...
+            {
+                if (entity.TryGetComponent<BoundingBoxComponent>(out var component))
+                {
+                    if (component.WorldAABB.Contains(position))
+                        yield return entity;
+                }
+                else
+                {
+                    var transform = entity.GetComponent<ITransformComponent>();
+                    if (position == transform.Position)
```
Comment (PJB3005) [inline SS14.Client/GameObjects/ClientEntityManager.cs]: No hard float equality checking, it's too unreliable.

File: SS14.Client/GameObjects/ClientEntityManager.cs [hunk 2]
```diff
@@ -53,6 +53,26 @@ public IEnumerable<IEntity> GetEntitiesIntersecting(Box2 position)
...
         }
 
+        public IEnumerable<IEntity> GetEntitiesIntersecting(Vector2 position)
+        {
+            foreach (var entity in _entities.Values)
+            {
+                if (entity.TryGetComponent<BoundingBoxComponent>(out var component))
+                {
+                    if (component.WorldAABB.Contains(position))
+                        yield return entity;
```
Comment (PJB3005) [inline SS14.Client/GameObjects/ClientEntityManager.cs:63]: Wrap this in braces. No brace-less if statements.

File: SS14.Client/GameObjects/Components/Renderable/WearableAnimatedSpriteComponent.cs [hunk 3]
```diff
@@ -43,6 +44,13 @@ public void SetCarriedSprite(string spritename)
             CarriedSprite = spritename;
         }
 
+        public override Sprite GetCurrentSprite()
+        {
+            if (!IsCurrentlyWorn)
```
Comment (PJB3005) [inline SS14.Client/GameObjects/Components/Renderable/WearableAnimatedSpriteComponent.cs:49]: No brace-less ifs.

File: SS14.Client/GameObjects/ClientEntityManager.cs (hunk 2 above)
Comment (clusterfack) [reply SS14.Client/GameObjects/ClientEntityManager.cs]: Sure we can make that a standard, but it makes no sense to start making it a requirement when nothing in the codebase follows it.

Like we always do make a PR to change it everywhere and then lets standardize it.

File: SS14.Client/GameObjects/ClientEntityManager.cs (hunk 1 above)
Comment (clusterfack) [reply SS14.Client/GameObjects/ClientEntityManager.cs]: Need a standard function for this

File: SS14.Client/GameObjects/Components/Renderable/WearableAnimatedSpriteComponent.cs (hunk 3 above)
Comment (clusterfack) [reply SS14.Client/GameObjects/Components/Renderable/WearableAnimatedSpriteComponent.cs]: Ditto above

**Comment by clusterfack** [comment]
The braces will end up getting changed once we PR the regexed brace changes

**Pull request merged**